modules/library/main/src/main/java/org/geotools/data/AbstractDataAccessFactory.java

```diff
@@ -0,0 +1,189 @@
+     * @param params
+     * @return true if params is in agreement with getParametersInfo, override for additional checks.
+     */
+    public boolean canProcess( Map params ) {
+        if (params == null) {
+            return false;
+        }
+        Param arrayParameters[] = getParametersInfo();
+        for (int i = 0; i < arrayParameters.length; i++) {
+            Param param = arrayParameters[i];
+            Object value;
+            if( !params.containsKey( param.key ) ){
+                if( param.required ){
+                    return false; // missing required key!
+                } else {
+                    continue;
+                }
+            }
+            try {
+                value = param.lookUp( params );
+            } catch (IOException e) {
+                // could not upconvert/parse to expected type!
+                // even if this parameter is not required
+                // we are going to refuse to process
+                // these params
+                return false;
+            }
+            if( value == null ){
+                if (param.required) {
+                    return (false);
+                }
+            } else {
+                if ( !param.type.isInstance( value )){
+                    return false; // value was not of the required type
+                }
+                if( param.metadata != null ){
+                    // check metadata
+                    if( param.metadata.containsKey(Param.OPTIONS)){
+                        List<Object> options = (List<Object>) param.metadata.get(Param.OPTIONS);
+                        if( options != null && !options.contains(value) ){
+                            return false; // invalid option
+                        }
+                    }
+                }
+            }
+        }
+        return true;
+    }
+    
+    /**
+     * Defaults to true, only a few datastores need to check for drivers.
+     *
+     * @return <code>true</code>, override to check for drivers etc...
+     */
+    public boolean isAvailable() {
+        return true;
+    }
+    
+    public ParameterDescriptorGroup getParameters(){
+        Param params[] = getParametersInfo();
+        DefaultParameterDescriptor parameters[] = new DefaultParameterDescriptor[ params.length ];
+        for( int i=0; i<params.length; i++ ){
+            Param param = params[i];
+            parameters[i] = new ParamDescriptor( params[i] );
+        }
+        Map properties = new HashMap();
+        properties.put( "name", getDisplayName() );
+        properties.put( "remarks", getDescription() );
+        return new DefaultParameterDescriptorGroup(properties, parameters);
+    }
+
+    /**
+     * Returns the implementation hints. The default implementation returns en empty map.
+     */
+    public Map<java.awt.RenderingHints.Key, ?> getImplementationHints() {
+        return Collections.EMPTY_MAP;
+    }
+}
+
+class ParamDescriptor extends DefaultParameterDescriptor {
```

Comment:
Not too sure about this, we do have:

* http://docs.geotools.org/latest/javadocs/org/geotools/data/DataAccessFactory.Param.html
* http://docs.geotools.org/latest/javadocs/org/geotools/data/Parameter.html

What would be the reason to use ParameterDescriptor and ParameterDescriptorGroup?

If we want to proceed this direction a change proposal will be needed (so we get a consistent vision and a migration plan).